### ISSUE TITLE
Is the calculated stopping distance too conservative?

### ISSUE THREAD

**Issue by PaulVerhoeckx** [nobleo/path_tracking_pid#146]
The stopping distance in the end phase `d_end_phase` ([here](https://github.com/nobleo/path_tracking_pid/blob/dbe631d210f389466cc3069651588aff1ce39818/src/controller.cpp#L448-L460)) contains a component which accounts for the distance traveled during the possible delay due to having a sampling time.

However I wonder if it isn't too conservative, currently it is:
```
target_x_vel * 2 * dt
```
Whereas I think the _average_ delay is `0.5*dt`, which would result in the following component:
```
target_x_vel * 0.5 * dt
```

**Comment by cesar-lopez-mar** [comment]
You are correct, however in practice this amount of compensation worked well. The need of compensation comes not only by the discrete sampling delay but also because of delays (and possible phase lag) in the robot's velocity controllers. Perhaps we should separate the compensation of both sources of delays.

**Comment by PaulVerhoeckx** [comment]
If this added compensation is robot specific (which I assume) it should at least be configurable but maybe we should even rethink the open-loop end phase of the velocity profile.

**Comment by PaulVerhoeckx** [comment]
I agree that controlling the forward velocity would introduce new configuration parameters, and that these should be kept to a minimum. However, I also wonder whether the open-loop solution with 1 parameter (delta t, as we discussed above) can model the braking distance very well for all `target_x_vel`'s. Because, as you said, different frictions play at different velocities. 

Have you tested the accuracy of the current brake distance implementation with different `target_x_vel`'s on a robot?
